contrib/hashicorp/vault/example_test.go

```diff
@@ -0,0 +1,95 @@
+package vault_test
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/contrib/hashicorp/vault"
+
+	"github.com/hashicorp/vault/api"
+)
+
+// This is the most basic way to enable tracing with Vault
+func ExampleNewHTTPClient() {
+	c, err := api.NewClient(&api.Config{
+		HttpClient: vault.NewHTTPClient(),
+		Address:    "http://vault.mydomain.com:8200",
+	})
+	if err != nil {
+		log.Fatalf("Failed to create vault client: %s\n", err)
+	}
+
+	// This call wil be traced
+	c.Logical().Read("/secret/key")
+}
+
+// Options can be passed in to configure the tracer
```

Comment:
Maybe something along the lines of:
```suggestion
// NewHTTPClient may also be called with additional customization.
```
We usually refer to the core library as the "tracer". Here we are configuring the integration.